backend/layers/processing/process_download.py

```diff
@@ -75,6 +81,41 @@ def remove_prefix(string: str, prefix: str) -> str:
         else:
             return string
 
+    def create_batch_job_definition_parameters(self, local_filename: str, dataset_id) -> Dict[str, Any]:
+        MEMORY_MODIFIER = 1.1  # add 10% overhead
+        MEMORY_PER_VCPU = 4000
+        MIN_MEMORY_MB = 4000
+        MIN_VCPU = 1
+        # The largest machine we are allocating is r5ad.2xlarge. This machine has 64GB of memory and 16 vCPUs.
+        MAX_MEMORY_MB = 64000
+        MAX_VCPU = 16
+        SWAP_MEMORY_MB = 300000
+
+        adata = scanpy.read_h5ad(local_filename)
+
+        # Note: this is a rough estimate of the uncompressed size of the dataset. This method avoid loading the entire
+        # dataset into memory.
+        uncompressed_size_MB = adata.n_obs * adata.n_vars // MB
```

Comment:
Why a factor of 4?

Reply:
Each element is an int64 (or a float32, for the normalized layer), so 4 bytes.

Reply:
Also, since this is likely to be a sparse matrix, this is probably an over-estimate anyway. If I recall correctly, the reason for loading the matrix into memory is to determine the sparsity - but iirc, curators aren't really updating dense matrices anymore. Therefore, we might have an opportunity to optimize even more. Maybe worth a discussion with them at some point.

Reply:
schema migration will automatically update everything in the corpus, so we do need to account for any dense matrices that are already in the corpus

Reply:
@Bento007 did you try rerunning your migration and allocating 4 bytes per value?

Separately, iff that works, then it might be worth, as a followup, taking into account the sparsity rather than assuming dense.

Reply:
I did not but I did find that using getsizeof to be a slightly better approach to estimating

Reply:
getsizeof applied to what exactly?